crates/text-parser/src/actions.rs

```diff
@@ -95,8 +95,8 @@ pub trait Actions {
         name: Option<&str>,
         wasm_ty: Self::WasmFuncTypeRef,
         webidl_ty: Self::WebidlTypeRef,
-        params: Self::OutgoingBindingMap,
-        result: Self::IncomingBindingMap,
+        params: Option<Self::OutgoingBindingMap>,
+        result: Option<Self::IncomingBindingMap>,
```

Comment:
Logically, a binding map is not complete without its params and results, its just that the tuple of values can be empty (and therefore the binding expressions are also empty).

We can already represent that today with a binding map's `bindings` being an empty vec, and it doesn't incur extra space overhead of the `Option`.

Let's remove this new option, and represent the lack of parameters/results with empty bindings.

Reply:
Ah, I see that this is only in the actions and not in the AST. This is fine, then I think.

Reply:
Correct :-).